src/main.rs

```diff
@@ -1,13 +1,47 @@
-use lc3_vm::command_line_parser::parser::{
-    execute_program_from_file, receive_command_line_arguments,
+use lc3_vm::{
+    command_line_parser::parser::receive_command_line_arguments,
+    virtual_machine_start::{
+        debug_program_from_file, execute_program_from_file, execute_vm_in_interactive_mode,
+    },
 };
+use termios::{tcsetattr, Termios, ECHO, ICANON, TCSANOW};
+
+const STDIN: i32 = 0;
+
+struct TermiosWrapper {
+    termios: Termios,
+}
+
+impl TermiosWrapper {
+    pub fn new(mut termios: Termios) -> Result<Self, String> {
+        termios.c_lflag &= !ICANON & !ECHO;
+
+        tcsetattr(STDIN, TCSANOW, &termios).map_err(|error| error.to_string())?;
+        Ok(Self { termios })
+    }
+}
```

Comment:
You have to 'save' the original `termios` unmodified in order to restore it in `drop`.
```suggestion
    pub fn new(termios: Termios) -> Result<Self, String> {
        let mut new_termios = termios;
        new_termios.c_lflag &= !ICANON & !ECHO;

        tcsetattr(STDIN, TCSANOW, &new_termios).map_err(|error| error.to_string())?;
        Ok(Self { termios })
    }
}
```

Reply:
Oh really? Didn't know about that one, thanks 🤝

Reply:
Just tried it and if It's a new termios is created, the code doesn't really works (I tried a lot of things and seems that the moment the struct its created, its also is dropped)

Reply:
Is it fixed now with [this commit](https://github.com/belfortep/lc3-vm/pull/2/commits/9b5a9c380d0dd3f500108d9e6e6e993132d6af57)?